planning/autoware_planning_validator/src/utils.cpp

```diff
@@ -146,45 +146,83 @@ std::pair<double, size_t> calcMaxCurvature(const Trajectory & trajectory)
 
-  const auto max_curvature_it = std::max_element(curvature_arr.begin(), curvature_arr.end());
-  const size_t index = std::distance(curvature_arr.begin(), max_curvature_it);
+  const auto max_curvature_it = std::max_element(curvature_vector.begin(), curvature_vector.end());
+  const size_t index = std::distance(curvature_vector.begin(), max_curvature_it);
 
   return {*max_curvature_it, index};
 }
 
+void calc_interval_distance(
+  const Trajectory & trajectory, std::vector<double> & interval_distance_vector)
+{
+  if (trajectory.points.size() < 2) {
+    interval_distance_vector = std::vector<double>(trajectory.points.size() - 1, 0.0);
+    return;
+  }
+
+  interval_distance_vector = std::vector<double>(trajectory.points.size() - 1, 0.0);
+  for (size_t i = 0; i < trajectory.points.size() - 1; ++i) {
+    const auto d = calc_distance2d(trajectory.points.at(i), trajectory.points.at(i + 1));
+    interval_distance_vector.at(i) = d;
+  }
+}
+
 std::pair<double, size_t> calcMaxIntervalDistance(const Trajectory & trajectory)
 {
   if (trajectory.points.size() < 2) {
     return {0.0, 0};
   }
+  std::vector<double> interval_distance_vector;
+  calc_interval_distance(trajectory, interval_distance_vector);
 
-  double max_interval_distances = 0.0;
-  size_t max_index = 0;
-  for (size_t i = 1; i < trajectory.points.size(); ++i) {
-    const auto d = calc_distance2d(trajectory.points.at(i), trajectory.points.at(i - 1));
-    if (max_interval_distances < std::abs(d)) {
-      takeBigger(max_interval_distances, max_index, std::abs(d), i);
-    }
+  const auto max_interval_it =
+    std::max_element(interval_distance_vector.begin(), interval_distance_vector.end());
+  const size_t max_index = std::distance(interval_distance_vector.begin(), max_interval_it);
+
+  return {*max_interval_it, max_index};
+}
+
+void calc_lateral_acceleration(
+  const Trajectory & trajectory, std::vector<double> & lateral_acceleration_vector)
+{
+  if (trajectory.points.size() < 2) {
+    lateral_acceleration_vector = std::vector<double>(trajectory.points.size(), 0.0);
+    return;
+  }
+
+  std::vector<double> curvature_vector;
+  calcCurvature(trajectory, curvature_vector);
+
+  lateral_acceleration_vector = std::vector<double>(trajectory.points.size(), 0.0);
+  for (size_t i = 0; i < trajectory.points.size(); ++i) {
+    const auto v_lon = trajectory.points.at(i).longitudinal_velocity_mps;
+    const auto a_lon = trajectory.points.at(i).acceleration_mps2;
+
+    // Component 1: Centrifugal acceleration from curvature (v^2 * κ)
+    const auto lat_acc_curve = v_lon * v_lon * curvature_vector.at(i);
+
+    // Component 2: Lateral projection of longitudinal acceleration
+    const auto theta = std::atan2(curvature_vector.at(i) * v_lon * v_lon, a_lon);
+    const auto lat_acc_from_lon = a_lon * std::sin(theta);
+
+    lateral_acceleration_vector.at(i) =
+      std::sqrt(lat_acc_curve * lat_acc_curve + lat_acc_from_lon * lat_acc_from_lon);
```

Comment:
```suggestion
      std::hypot(lat_acc_curve  lat_acc_from_lon);
```

Reply:
fixed in https://github.com/autowarefoundation/autoware_universe/pull/10385/commits/7ebe903217fb64b26df9466c4bda0ef66a94955e